tools/scope_manage_annotation.py

```
@@ -0,0 +1,178 @@
+
+    # Initial checks for origin and key, always necessary
+    if origin is None:
+        print('Error: please specify origin to %s' % action)
+        return
+
+    if key is None:
+        print('Error: please specify key to %s' % action)
+        return
+
+    # check if source is single object or csv file of many
+    if source[-4:] == '.csv':
+        file = pd.read_csv(source)  # modify input formats to prepare for loop
+        obj_ids = file['obj_id']
+        if (action == 'update') | (action == 'post'):
+            values = file[key]
+    else:
+        obj_ids = [source]  # modify single source input formats to prepare for loop
+        if value is not None:
+            values = [float(value)]
+        else:
+            values = [value]
+
+    # loop over objects, performing specified annotation action
+    for i in range(len(obj_ids)):
+        obj_id = obj_ids[i]
+
+        # update and delete branches require GET for existing annotation
+        if (action == 'update') | (action == 'delete'):
+            matches = 0
+            # get all annotations for object
+            response = api("GET", '/api/sources/%s/annotations' % obj_id, token).json()
+            data = response.get('data')
+
+            # loop through annotations, checking for match with input key and origin
+            for entry in data:
+                annot_id = str(entry['id'])
+                annot_origin = entry['origin']
+                annot_data = entry['data']
+
+                annot_name = [x for x in annot_data.keys()][0]
+                annot_value = [x for x in annot_data.values()][0]
+
+                # if match is found, perform action
+                if (key == annot_name) & (origin == annot_origin):
+                    matches += 1
+                    if action == 'update':
+                        value = values[i]
+
+                        # Check value if performing update or post actions
+                        if value is None:
```

Maybe raise an error here?